lib/shrine/plugins/download_endpoint.rb

```diff
@@ -65,14 +65,20 @@ def initialize(file)
           @file = file
         end
 
-        def call(host: self.host)
-          [host, *prefix, path].join("/")
+        def call(host: self.host, expires_in: nil)
+          [host, *prefix, path(expires_in: expires_in)].join("/")
         end
 
         protected
 
-        def path
-          file.urlsafe_dump(metadata: %w[filename size mime_type])
+        def path(expires_in: nil)
+          if expires_in || default_expires_in
+            reference = file.urlsafe_dump(metadata: %w[filename size mime_type])
+            expires_at = Time.now + (expires_in || default_expires_in)
```

Comment:
Not 100% sure what to do here, I am used to ActiveSupports monkeypatch rails methods.
Maybe the argument should not be `expires_in` and instead be `expires_at` to avoid having to handle timezones and such.

Reply:
The expiration timestamp should probably be in integer (aka UNIX) format in the URL, that way you don't have to worry about time zones.